src/main/scala/beam/agentsim/agents/household/HouseholdCAVScheduling.scala

```diff
@@ -0,0 +1,195 @@
+package beam.agentsim.agents.household
+import beam.agentsim.agents.planning.BeamPlan
+import org.matsim.api.core.v01.Coord
+
+import org.matsim.api.core.v01.population._
+
+import scala.collection.immutable.Map
+import scala.collection.immutable.List
+
+sealed trait MobilityServiceRequestType
+case object MSRPickup extends MobilityServiceRequestType
+case object MSRDropoff extends MobilityServiceRequestType
+case object MSRRelocation extends MobilityServiceRequestType
+case object MSRInit extends MobilityServiceRequestType
+
+case class MobilityServiceRequest(
+  person: Option[Person],
+  activity: Activity,
+  time: Double,
+  deltaTime: Double,
+  tag: MobilityServiceRequestType
+) {
+  override def toString =
+    s"$tag{ ${person match {
+      case Some(x) => x.getId
+      case None    => "NA"
+    }}|${activity.getType}|${(time / 3600).toInt}:${((time % 3600) / 60).toInt}:${(time % 60).toInt}|ocp:$deltaTime }"
+}
+
+class HouseholdPlansToMSR(plans: List[BeamPlan]) {
+  var requests = List[MobilityServiceRequest]()
+  for (plan <- plans) {
+    for (activity <- plan.activities) {
+      if (!activity.getStartTime.isInfinity && !activity.getStartTime.isNaN)
+        requests = requests :+ new MobilityServiceRequest(
+          Some(plan.getPerson),
+          activity,
+          activity.getStartTime,
+          0.0,
+          MSRDropoff
+        )
+      if (!activity.getEndTime.isInfinity && !activity.getEndTime.isNaN)
+        requests = requests :+ new MobilityServiceRequest(
+          Some(plan.getPerson),
+          activity,
+          activity.getEndTime,
+          0.0,
+          MSRPickup
+        )
+    }
+  }
+  requests = requests.sortWith(_.time < _.time)
+  def apply(): List[MobilityServiceRequest] = { requests }
```

Comment:
This is an odd use of `apply`, which is usually meant as a de-sugaring method. But here it seems as though it should be more of a property accessor. So maybe it should be renamed to `requests`, so that it can be use akin to `householdplans.requests` - in fact I would just turn the whole thing into one expression which could be a `lazy val requests = for-comprehension`